qiskit/transpiler/passes/layout/layout_2q_distance.py

```diff
@@ -52,12 +52,19 @@ def run(self, dag):
         if layout is None:
             return
 
+        if self.coupling_map is None or len(self.coupling_map.graph) == 0:
+            self.property_set[self.property_name] = 0
+            return
+
+        self.coupling_map.compute_distance_matrix()
+
         sum_distance = 0
 
+        virtual_physical_map = layout.get_virtual_bits()
         for gate in dag.two_qubit_ops():
-            physical_q0 = layout[gate.qargs[0]]
-            physical_q1 = layout[gate.qargs[1]]
+            physical_q0 = virtual_physical_map[gate.qargs[0]]
+            physical_q1 = virtual_physical_map[gate.qargs[1]]
 
-            sum_distance += self.coupling_map.distance(physical_q0, physical_q1) - 1
+            sum_distance += self.coupling_map._dist_matrix[physical_q0, physical_q1] - 1
```

Comment:
Can the `self.coupling_map._dist_matrix` similarly be moved outside the loop and converted to a `self.coupling_map.distance_matrix`, or is it needed for performance? If so, can you add a comment?

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/4e4da2b8d072705c4d53f7930f0866fb2115012a